dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/DownloadInstanceWriter.java

```diff
@@ -350,6 +350,20 @@ private void writeStorageIOToOutputStream(StorageIO<DataFile> storageIO, OutputS
         }
     }
 
+    private static String encodeURI(String string) throws UnsupportedEncodingException {
+        if (string == null) {
+            return null;
+        }
+
+        return URLEncoder.encode(string, "UTF-8")
```

Comment:
Ok I have digged a lot and well filename encoding in browsers are a mess. I don't think we should manually replace characters as we please without being compatible with encoding documentation. Also here we are encoding and decoding some characters which is kinda messy. I found a solution which is compatible with official rfc5987 which defines how to encode characters in http headers. 
```
fileName = MimeUtility.encodeText(fileName);

httpHeaders.add("Content-Disposition", "attachment; filename="+fileName + "");
```
This should work.

MimeUtility is created according to their RFC but Http headers encoding RFC is based on MIME RFC :P.

Reply:
This is not 'replacing characters as we please'. This solution is directly taken from fix to primefaces: https://github.com/primefaces/primefaces/pull/2368 and is compatible with RFC6266

Reply:
Ah ok, then it should be mentioned in javadoc, since without context I have no clue why those specific characters are replaced.